shared/actions/git.js

```diff
@@ -234,45 +131,72 @@ function _handleIncomingGregor(action: GitGen.HandleIncomingGregorPayload) {
   }
 }
 
-function _navigateToTeamRepo(action: GitGen.NavigateToTeamRepoPayload) {
-  return Saga.sequentially([
-    Saga.call(_loadGit, GitGen.createLoadGit()),
-    Saga.identity(action),
-    // This needs to be a select so we get the store post-loadGit
-    Saga.select((state: TypedState) => state.entities.getIn(['git', 'idToInfo'])),
-  ])
+const navToGit = (_, action: GitGen.NavToGitPayload) => {
+  const path = isMobile ? [Tabs.settingsTab, SettingsConstants.gitTab] : [Tabs.gitTab]
+  const actions = [Saga.put(navigateTo(path, []))]
+  if (action.payload.routeState) {
+    actions.push(Saga.put(setRouteState(path, action.payload.routeState)))
+  }
+  return Saga.all(actions)
 }
 
-function _processNavigateToTeamRepo(results: any[]) {
-  const {teamname, repoID} = (results[1]: GitGen.NavigateToTeamRepoPayload).payload
-  const idToInfo: I.Map<string, Types.GitInfo> = results[2]
+const isRepoInfoFresh = (lastLoad: ?number, maxAgeMs: number) => lastLoad && lastLoad - Date.now() < maxAgeMs
+
+// Note: Needs to be sequential (& possibly refire) in order to match the repoID + teamname to a unique id
+const navigateToTeamRepo = (state: TypedState, action: GitGen.NavigateToTeamRepoPayload) => {
```

Comment:
This was a little tricky since it requires data from another action. I went with calling the other action and refiring this action when the data is available. If there's a better solution lmk!

Reply:
I guess another solution would be for the expandedSet to be either the unique `id` or `{repoId: string, teamname: string}`, then derive the unique id in the connector. I'm not convinced that's better, but it would avoid having to do the tricky sequential put/take and refire. Thoughts?